src/components/Code/Code.tsx

```diff
@@ -0,0 +1,125 @@
+import { CopyIcon, CheckIcon } from "@chakra-ui/icons";
+import { Box, Flex, IconButton, useClipboard } from "@chakra-ui/react";
+import Highlight, { defaultProps, Language } from "prism-react-renderer";
+import nightOwlLight from "prism-react-renderer/themes/nightOwlLight";
+import { ComponentPropsWithoutRef, FunctionComponent } from "react";
+
+interface CodeProps {
+  code: string;
+  language: Language;
+}
+
+// Unfortunately not exported by prism-react-renderer
+type RenderProps = Parameters<
+  ComponentPropsWithoutRef<typeof Highlight>["children"]
+>[0];
+
+const btnThemes = {
+  singleLine: {
+    colorScheme: "blue",
+    color: "blue.500",
+    ml: 4,
+  },
+  multiLine: {
+    color: "white",
+    sx: {
+      ":hover": {
+        bg: "blue.400",
+      },
+    },
+  },
+};
+
+const CopyButton: FunctionComponent<{
+  value: string;
+  isMultiLine?: boolean;
+}> = ({ isMultiLine, value }) => {
+  const { hasCopied, onCopy } = useClipboard(value);
+  const theme = isMultiLine ? btnThemes.multiLine : btnThemes.singleLine;
+
+  return (
+    <IconButton
+      aria-label="Copy Code"
+      h={6}
+      icon={hasCopied ? <CheckIcon color="green.200" /> : <CopyIcon />}
+      minW="auto"
+      onClick={onCopy}
+      variant="ghost"
+      w={6}
+      {...theme}
+    />
+  );
+};
+
+const SingleLineRenderer: FunctionComponent<RenderProps & { code: string }> = ({
```

Comment:
These renderers will probably also get their own files, and the syntax highlighting code will become a general component